packages/story-editor/src/components/publishModal/mainContent/mandatoryStoryInfo.js

```diff
@@ -42,15 +45,55 @@ const _TextArea = styled(TextArea)`
   margin: 8px 0;
 `;
 
-const MandatoryStoryInfo = ({
-  handleUpdateStoryInfo,
-  inputValues: _inputValues,
-}) => {
-  const [inputValues, setInputValues] = useState(_inputValues);
+const MandatoryStoryInfo = () => {
+  const storyId = useConfig(({ storyId }) => storyId);
+
+  const updateStory = useStory(({ actions }) => actions.updateStory);
+  const _inputValues = useStory(({ state: { story } }) => ({
+    [INPUT_KEYS.EXCERPT]: story.excerpt,
+    [INPUT_KEYS.TITLE]: story.title || '',
+  }));
+  // Keep slug isolated from other input values to limit changes and keep in sync on handleSlugUpdate
+  const slug = useStory(
+    ({
+      state: {
+        story: { slug },
+      },
+    }) => slug
+  );
```

Comment:
this is all cut from `publishModal` since it's only needed here to avoid passing around props. 

The only thing that i changed is isolating `slug` from `_inputValues` to keep the value separated for permalink updating.